bokeh/io.py

```diff
@@ -442,10 +442,10 @@ def _get_save_args(state, filename, resources, title):
 def _save_helper(obj, filename, resources, title, validate):
     remove_after = False
     if isinstance(obj, LayoutDOM):
-        doc = obj.document
-        if doc is None:
-            doc = Document().add_root(obj)
```

Comment:
The problem was here, in that `Document.add_root` doesn't return a value. So trying the call `validate` on the `None` value was raising an exception and killing the integration tests.